src/primitives/block.cpp

```diff
@@ -109,6 +112,47 @@ uint256 CBlock::CheckMerkleBranch(uint256 hash, const std::vector<uint256>& vMer
     return hash;
 }
 
+static uint64_t next_pow2(uint64_t x)
```

Comment:
This does not return the next power of two for input x = 0. Also it should be documented where this comes from: https://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2

Reply:
It shouldn't occur; a block will always contain at least a coinbase transaction.

Reply:
Also, it's actually fine that on input `x = 0` this function returns `0`. That would mean we "pad" an empty tree to an empty tree, which is handled at the end of `CBlock::BuildAuthDataMerkleTree` by returning the all-zeroes root.

Reply:
Ah, looks like it is intended to return 0 for x = 0.